### PR TITLE
implement API for EvaluationResult protocol

### PR DESCRIPTION
**Pull request by yuki-mt**
## What is this PR for?

add method for EvaluationResult protocol to return detailed evaluation result to a client.
related to https://github.com/rekcurd/drucker/pull/24

Change in frontend will be the next PR.

## This PR includes

- move API code about evaluation to `api_evaluation.py`
    - also, move test code to `drucker_dashboard/test/test_api_evaluation.py`
- add response marshalling to evaluation APIs.
- update grpc submodule to rekcurd/drucker-grpc-proto#13
- add API and core method for EvaluationResult protocol
- add `f.seek(0)` and reset iteration in hash_util.py to prevent StopIteration Error due to more file reading.

## What type of PR is it?

Feature, Bugfix

## What is the issue?

N/A

## How should this be tested?

`python -m unittest test/test_api_evaluation.py`

### CODE REVIEW

**Comment by yuki-mt** [comment]
Probably this PR has conflict with https://github.com/rekcurd/drucker-dashboard/pull/54 , so I will fix this PR and remove WIP after merging #54

File: app/models/evaluation_result.py [hunk 1]
```diff
@@ -21,9 +22,23 @@ class EvaluationResult(db.Model):
...
+        res['result_id'] = self.evaluation_result_id
+        return res
+
+    @result.setter
+    def result(self, value):
+        if isinstance(value, dict):
+            res = json.dumps(value)
+        else:
+            res = value
+        self._result = res
```
Comment (yuki-mt) [inline app/models/evaluation_result.py]: This is to

- get `result` as dict in Python while storing in db as string.
- `result` contains `result_id` in the dict

**Comment by yuki-mt** [comment]
I resolved conflict. Please review it

**Comment by yuki-mt** [comment]
we need the same version of drucker-grpc between dashboard and client to pass e2e test...

So we need to do

1. release https://github.com/rekcurd/drucker/pull/24
2. update `drucker`, merge, and release https://github.com/rekcurd/drucker-client/pull/14
3. update `drucker_client` in this PR

**Comment by yuki-mt** [comment]
I updated `drucker-client` and fix e2e test. Please review it now

File: drucker_dashboard/apis/api_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        if eval_query.one_or_none() is None:
+            return {"status": False, "message": "Not Found."}, 404
+
+        eval_query.delete()
+        db.session.query(EvaluationResult)\
+            .filter(EvaluationResult.evaluation_id == evaluation_id).delete()
+        db.session.commit()
+        db.session.close()
+
+        return {"status": True, "message": "Success."}
```
Comment (keigohtr) [inline drucker_dashboard/apis/api_evaluation.py]: **Reminder**
We need to delete the file also.

File: drucker_dashboard/apis/api_evaluation.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    eval_parser.add_argument('evaluation_id', location='form', type=int, required=False)
+    eval_parser.add_argument('overwrite', location='form', type=bool, required=False)
+
+    @eval_info_namespace.expect(eval_parser)
+    @eval_info_namespace.marshal_with(eval_metrics)
+    def post(self, application_id:int):
+        """evaluate"""
+        args = self.eval_parser.parse_args()
+        eval_id = args.get('evaluation_id', None)
+        service_id = args.get('service_id')
```
Comment (keigohtr) [inline drucker_dashboard/apis/api_evaluation.py]: The combination of `evaluation_id` and `model_id` will be better for UX.
To realize it, we need some features but at the moment how about the following spec?

1. Parameters are `evaluation_id` and `model_id` (I prefer to use path parameter for `id` but it doesn't matter)
1. Raise errors when specifying `model_id` which are not used on services.
1. Raise errors when specifying production services.

File: drucker_dashboard/apis/api_evaluation.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                robj.result = result
+            db.session.flush()
+            response_body = robj.result
+            db.session.commit()
+            db.session.close()
+
+        return response_body
+
+
+@eval_info_namespace.route('/<int:application_id>/evaluation_result/<int:eval_result_id>')
```
Comment (keigohtr) [inline drucker_dashboard/apis/api_evaluation.py]: `evaluation_results` will be better.

File: drucker_dashboard/apis/api_evaluation.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+            .filter(Evaluation.application_id == application_id,
+                    EvaluationResult.evaluation_id == Evaluation.evaluation_id,
+                    EvaluationResult.evaluation_result_id == eval_result_id).one_or_none()
+        if eval_with_result is None:
+            return {"status": False, "message": "Not Found."}, 404
+
+        db.session.query(EvaluationResult)\
+            .filter(EvaluationResult.evaluation_result_id == eval_result_id).delete()
+        db.session.commit()
+        db.session.close()
```
Comment (keigohtr) [inline drucker_dashboard/apis/api_evaluation.py]: **Reminder**
We need to delete the file.

File: drucker_dashboard/apis/api_evaluation.py (hunk 3 above)
Comment (keigohtr) [reply drucker_dashboard/apis/api_evaluation.py]: Great work!

File: drucker_dashboard/models/evaluation_result.py [hunk 6]
```diff
@@ -20,9 +21,23 @@ class EvaluationResult(db.Model):
     model_id = Column(Integer, nullable=False)
     data_path = Column(String(512), nullable=False)
     evaluation_id = Column(Integer, ForeignKey('evaluations.evaluation_id'), nullable=False)
-    result = Column(Text, nullable=False)
+    _result = Column(Text, nullable=False)
```
Comment (keigohtr) [inline drucker_dashboard/models/evaluation_result.py]: Could you make an Issue about this field?
Storing json value is weird for me but I don't have a good idea :(

**Pull request merged**